From b3287343bd8870538aee70687286d78881ee8b81 Mon Sep 17 00:00:00 2001
From: rguenth <rguenth@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 31 Mar 2011 10:00:44 +0000
Subject: [PATCH] 2011-03-31  Richard Guenther  <rguenther@suse.de>

	PR lto/48246
	* lto.c (lto_wpa_write_files): Disable assert for non-empty
	partitions when checking is not enabled.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_6-branch@171774 138bc75d-0d04-0410-961f-82ee72b054a4

index 71c8cdb..4c832fb 100644
--- a/gcc/lto/lto.c
+++ b/gcc/lto/lto.c
@@ -1514,8 +1514,8 @@ lto_wpa_write_files (void)
 	  fprintf (cgraph_dump_file, "varpool nodes:");
 	  dump_varpool_node_set (cgraph_dump_file, vset);
 	}
-      gcc_assert (cgraph_node_set_nonempty_p (set)
-		  || varpool_node_set_nonempty_p (vset) || !i);
+      gcc_checking_assert (cgraph_node_set_nonempty_p (set)
+			   || varpool_node_set_nonempty_p (vset) || !i);
 
       lto_set_current_out_file (file);
 
-- 
1.7.0.4

